VG Music Studio/Core/GBA/MP2K/Player.cs

```diff
@@ -1257,7 +1294,20 @@ private void ExecuteNext(Track track, ref bool update)
                     }
                     case 0xB2:
                     {
-                        track.DataOffset = (_config.ROM[track.DataOffset++] | (_config.ROM[track.DataOffset++] << 8) | (_config.ROM[track.DataOffset++] << 16) | (_config.ROM[track.DataOffset++] << 24)) - GBA.Utils.CartridgeOffset;
+                        int jumpOffset = (_config.ROM[track.DataOffset++] | (_config.ROM[track.DataOffset++] << 8) | (_config.ROM[track.DataOffset++] << 16) | (_config.ROM[track.DataOffset++] << 24)) - GBA.Utils.CartridgeOffset;
+                        // HACK: only do jumps if we are jumping forwards
+                        // means we can't get into an infinite jump routine
+                        if (jumpOffset > track.DataOffset) {
```

Comment:
Wonder if this is worth keeping in here? Feels like this hack might only work for some games where there are no complex jump routines?

Reply:
removed